### PR TITLE
Cryptol/pr1136

### PR DESCRIPTION
**Pull request by robdockins**
Changes to track Cryptol PRs GaloisInc/cryptol#1048 and GaloisInc/cryptol#1136

Overlaps somewhat with #197, maybe it makes sense to rebase them together.

### CODE REVIEW

**Comment by robdockins** [comment]
@yav, following this update, one of the SAW integration tests is failing with the following message:

```
[19:38:02.144] Cryptol error:
[error]
    Overlapping symbols defined:
    (at Cryptol:844:11--844:17, Cryptol::update)
    (at cryptol-spec/mul_java.cry:7:1--7:7, mul_java::update)
```

I think this should probably be a shadowing warning instead.  If I just load up the module in the Cryptol REPL, it works, so I think maybe I messed up something subtle here in the `CryptolEnv` module.  Any ideas what might be the problem?

File: cryptol-saw-core/src/Verifier/SAW/CryptolEnv.hs [hunk 1]
```diff
@@ -544,18 +548,12 @@ parseDecls sc env input = do
...
     let topdecls = [ P.Decl (P.TopLevel P.Public Nothing d) | d <- npdecls ]
 
-    -- Label each TopDecl with the "interactive" module for unique name generation
-    let (mdecls :: [MN.InModule (P.TopDecl P.PName)]) = map (MN.InModule interactiveName) topdecls
-    nameEnv1 <- MN.liftSupply (MN.namingEnv' mdecls)
-
     -- Resolve names
-    let nameEnv = nameEnv1 `MR.shadowing` getNamingEnv env
-    (rdecls :: [P.TopDecl T.Name]) <- MM.interactive (MB.rename interactiveName nameEnv (traverse MR.rename topdecls))
+    (_nenv, rdecls) <- MM.interactive (MB.rename interactiveName (getNamingEnv env) (MR.renameTopDecls interactiveName topdecls))
```
Comment (yav) [inline cryptol-saw-core/src/Verifier/SAW/CryptolEnv.hs:552]: @robdockins I am  not really sure, but perhaps there is a problem here.   Note that in the old code the naming environment in which we do the resolving already has the names added to it, while in the new one it does not...

Otoh, I don't understand why the old code was doing that in the first place, and what you've replaced it with makes sense to me, so perhaps the problem is not here.

To be more help, I should probably just check out the code and run it, ping me on the chat if that'd be useful.

Comment (robdockins) [reply cryptol-saw-core/src/Verifier/SAW/CryptolEnv.hs:552]: I wondered about this code too, but I think it's unrelated, actually.  I think this code is for dealing with declarations appearing inside the `{{ ... }}` delimiters in SAWScript.  The code for importing modules is further above, and is barely changed.

**Comment by robdockins** [comment]
I think the Cryptol issue holding this up has been resolved, I plan to migrate this into the corresponding saw-script PR https://github.com/GaloisInc/saw-script/pull/1191

**Comment by robdockins** [comment]
These changes are now folded into GaloisInc/saw-script#1191

**Pull request closed without merging**